testing/test_mark.py

```diff
@@ -557,14 +570,14 @@ def test_has_inherited(self):
 
         """
         )
-        items, rec = testdir.inline_genitems(p)
+        items, rec = pytester.inline_genitems(p)
         has_own, has_inherited = items
-        assert has_own.get_closest_marker("c").kwargs == {"location": "function"}
-        assert has_inherited.get_closest_marker("c").kwargs == {"location": "class"}
+        assert has_own.get_closest_marker("c").kwargs == {"location": "function"}  # type: ignore
```

Comment:
When mypy emits an error, it also emits an error code, like `attr-defined` or such. We usually specify it in the `type: ignore` comment, like this: `type: ignore[attr-defined]`. This has two advantages: 1) it helps the reader understand what is the problem being ignored, and 2) if the error changes unexpectedly, it will start failing again, which is good.

Reply:
Thanks, I was kinda banging my head with this mypy stuff, get_cloest_marker returns an Optional[Mark] but we are calling .kwargs (None may not have it obviously) I suppressed but what is the correct approach to this scenario here?

Reply:
The common way to appease mypy is to use `assert is not None`, in this case

```py
marker = assert has_own.get_closest_marker("c")
assert marker is not None
assert marker.kwargs = ....
```